sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -171,14 +171,15 @@ object DataType {
   private val otherTypes = {
     Seq(NullType, DateType, TimestampType, BinaryType, IntegerType, BooleanType, LongType,
       DoubleType, FloatType, ShortType, ByteType, StringType, CalendarIntervalType,
-      DayTimeIntervalType, YearMonthIntervalType)
+      DayTimeIntervalType)
       .map(t => t.typeName -> t).toMap
   }
 
   /** Given the string representation of a type, return its DataType */
   private def nameToType(name: String): DataType = {
     name match {
       case "decimal" => DecimalType.USER_DEFAULT
+      case "interval year to month" => YearMonthIntervalType()
```

Comment:
shall we follow `FIXED_DECIMAL` and write a regex to support different units?

Reply:
This is just stub to pass compiler errors. I don't think manually parsing is the right approach. I do believe that:
- Having two type parsers is bad idea
- We should use existing SQL parser.

Reply:
I'm happy to see a refactor here to use the SQL parser instead.